### ISSUE TITLE
Plug and play voting system (Developer needed)

### ISSUE THREAD

**Issue by ghost** [EthereumCommonwealth/Proposals#84]
Idea, A third use case for $CLO:

What if anyone would like to make a vote for something, in their company, or sports team or something. 
They would google make a voting online.  Here Callisto network popps up! 

I would like to see a system where a creator of the voting can set a certain number or white-listed addresses that are sent randomly to user emails. ( The creator of the voting does not know who own what address and therefor what he votes for)
Creator only knows that the correct people gets the addresses who can vote on the subjects that he creates

The creator of the vote simply just needs enough CLO for the users to talk to the smart contract ( extremely small amounts of money)

Optimal implementation:
Step1: Creator of voting buy a bit CLO
Step2 Creator makes a voting (sets how many users are going to participate, adds all of the emails for those users, and creates the question and alternatives for the voting)

Step 3: The system automatically creates 1 whitelisted ( only whitelisted address can vote) addresse for every email address, sends them out randomly ( so the creator does not know who owns what adress) 

step 4: The system automatically sends enough CLO to every address for the user to participate in the voting 

This voting solution could be promoted like the most secure anonymous voting system in the world. 
Deployable in seconds for penny's
We are aiming for a decentralized future in the space of blockchain and cryptocurrency. The more this thinking evolves, the more voting will be implemented in all kinds of situations.  

This with a good UX(User experience) would be great for adoption of CLO 
Also if the creator could buy CLO with Visa or Mastercard here would be optimal ofc. (should be thought about how we can make happen) 

Would this be interesting and do we have a developer who can create this? 

______________________________________________________________________
Система голосования P&P (требуется разработчик)
 
Идея:
Когда кто-либо захочет создать голосование, в своей компании, либо спорти вной команде, или еще где либо, они, возможно, будут искать в интернете "запуск голосования онлайн". И, допустим, тут появляется Callisto Network!

Было бы круто увидеть систему, в которой создатель голосования может установить определенное количество голосующих, либо разослать "белый список" голосующих адресов на электронные почты пользователей. (Создатель не знает кому приналдлежат эти адреса, а знает только то, что целевая группа людей получит список с адресами тех, кто может голосовать, и соответственно представители этой целевой группы узнают, есть ли у них доступ к голосованию) 

Создателю голосования требуется лишь небольшое количество CLO, чтобы пользователи могли взаимодействовать со смартконтрактом (крайне маленькое количество денег)

Оптимальная реализация:

Шаг1: Создатель голосования покупает небольшое количество CLO

Шаг2: Создатель создает голосование (устанавливает количество голосующих, добавляет адреса e-mail этих пользователей, устанавливает вопрос и варианты ответов этого голосования).

Шаг3: Система автоматически создает по 1 адресу "белого списка голосователей" на каждый адрес e-mail (только адреса из белого списка могут участвовать в голосовании), и в запускает автоматическую рассылку в случайном порядке (чтобы создатель не знал какой адрес привязан к определенному адресу e-mail). 

Шаг 4: Система автоматически рассылает CLO на каждый адрес из белого списка, чтобы пользователи могли принять участие в голосовании.

Считаю, что подобную систему голосования можно продвигать, как самую безопасную анонимную систему голосования в мире. 
Которую можно запустить очень дешего

Мы стремимся к децентрализованному будущему в этом пространстве, и чем больше будет развиваться это мышление, тем больше будет голосов во всех ситуациях.

Если к этой системе добавить удобный пользовательский интерфейс, это будет способствовать распространению CLO
Также, если бы у создателя была возможность в этом же месте купить CLO картами Visa или Mastercard было бы неплохо. (Это стоило бы обдумать)

Как считаете, стоящая ли идея? И есть ли разработчик, кто может создать подобную систему?

![wooote](https://user-images.githubusercontent.com/39714534/68484605-7f456d00-0235-11ea-8b17-9b3ef5f314d0.png)

**Comment by shivneja** [comment]
Why limit this idea to private use? Why not expand this idea also to local community elections or any public voting or referendum? The only thing that needs to be kept in mind is that the vote needs to be secret.

edit: for development if you can give this to some university to implement as a project it could be a good starting point. Could also bring in some fresh talent into the CLO network who can learn from the experts. Would be very interesting to pursue this project.

**Comment by ghost** [comment]
> Why limit this idea to private use? Why not expand this idea also to local community elections or any public voting or referendum? The only thing that needs to be kept in mind is that the vote needs to be secret.
> 
> edit: for development if you can give this to some university to implement as a project it could be a good starting point. Could also bring in some fresh talent into the CLO network who can learn from the experts. Would be very interesting to pursue this project.

Agreed! That would be amazing if we could take it further and implement is as you say, 
Just thinking that more investigations of how we insure that only the people who are suppose to be able to vote can vote. Giving out voting rights to an email address given by the voter himself does not feel enough when we are talking about life changing public votes with millions of voters-
Will ID verification then be needed? and how is that then implemented in a secure and decentralized way? 

Thank you for response!

**Comment by shivneja** [comment]
> Just thinking that more investigations of how we insure that only the people who are suppose to be able to vote can vote. Giving out voting rights to an email address given by the voter himself does not feel enough when we are talking about life changing public votes with millions of voters-
> Will ID verification then be needed? and how is that then implemented in a secure and decentralized way?
> 
> Thank you for response!

Yes, the identity verification is an interesting challenge. How about going cross-chain? CLO is now a chainlink Hub, so it could be (theoretically) possible to delegate identity verification to one of the smart contracts on other chains specialising in Digital identity (I think Metaverse or DNA may be one?). In this way, the voting process and digital identity are kept independent and therefore more secure. The digital identity itself may require more than an email address but in a small community this should be easi(er) to address. 

Edit: if you wish to make this reality, converting it into a project to obtain resources should not be very difficult. Voting is a core need of any democratic establishment. So the possibilities are endless.

**Comment by Dexaran** [comment]
I would say that I like the idea but I don't see any real use case in it.

The idea proposes a voting system that will be based on CLO. The voting system gives each participant (email or identity) one vote. We have already working alternatives in the world. For example twitter polls.

The disadvantages of the proposed CLO-based voting system are:


1. It is not user-friendly.  This involves private keys, interaction with smart-contracts and a lot of things that are much harder than pushing a button in twitter.

2. It is not free.

3. This requires a centralized server that will generate and send out the private keys. The presence of a server negates the advantages of decentralization completely. 


At the other hand, I don't see even a single advantage of this voting system compared to the above mentioned twitter polls. This is not "the most secure" voting system if we have a server component responsible for something in this system.

**Comment by shivneja** [comment]
I differ, I think the idea proposed is a good one depending on the ultimate use, @Spigfish can confirm if my understanding is correct.

A poll (as a twitter poll) would be used for a query such as “do you prefer white creamy pasta sauce or do you prefer red arabbiata pasta sauce” and the audience would choose between these 2 options to decide the winner. This poll is random and public.

However, a vote, for example to choose the best science project which gets awarded a prize would need to be secret, but yet identifiable and stored in archives for long. The same would apply for example for electing a local student body or government or within a company that wants to decide which new innovation to take forward. Even CLO community could implement this to decide which github project should be supported! 

If @Spigfish is proposing “vote” then this project is worth looking at.
> 
> The disadvantages of the proposed CLO-based voting system are:
> 
> 1. It is not user-friendly.  This involves private keys, interaction with smart-contracts and a lot of things that are much harder than pushing a button in twitter.

I guess, if the proposal is intended to be a “Voting” system, it needs to be more difficult. 

> 2. It is not free.

Should not be free. Pay to play. Ultimately there is no free lunch. Twitter is selling your data anyways, which is perfect for random polls, but data about a secret vote should not be sold.

> 3. This requires a centralized server that will generate and send out the private keys. The presence of a server negates the advantages of decentralization completely.
> 

> At the other hand, I don't see even a single advantage of this voting system compared to the above mentioned twitter polls. This is not "the most secure" voting system if we have a server component responsible for something in this system.

The weakness of twitter is bots (refer Jack Dorsey discussion with Elon Musk “Elon Musk’s suggestion for how to fix Twitter? Identify the bots.”), and the strength of CLO is it’s security. The identification process is already being addressed in certain geographies, see the MyID project in Korea or DNA/ETP in China. In the next wave I believe that crypto users will get stamped and identified with all the regulatory attention. In any “Voting” system actually identification of the Voter is necessary but identification of the “Vote” should be secret.

**Comment by Dexaran** [comment]
@shivneja Well, you convinced me that this proposal makes sense.

However the issue of centralization still stands: 

> 3. This requires a centralized server that will generate and send out the private keys. The presence of a server negates the advantages of decentralization completely.

Centralization of power is not that beneficial for the project in general. In case of a voting system this is also a major security flaw.

**Comment by Dexaran** [comment]
I guess that the process could be tuned to eliminate the requirement of a centralized server:

Optimal implementation:
Step 1: The creator of voting is buying some CLO.

Step 2: The creator makes voting (sets how many users are going to participate, adds all the `ADDRESSES` for those users, and creates the question and options for the voting).

Step 3: The system automatically whitelists the `ADDRESSES` proposed at Step 2. ( only whitelisted address can vote).

Step 4: The system automatically sends CLO enough for the user to participate in the voting to every whitelisted address.

This requires users to create addresses and ensure privacy (by setting a set of rules for the voting that will prevent them from knowing who owns which address) at their own. Unfortunately, the smart-contract can not handle this in a proper way so the only "secure" approach is to delegate privacy/address generation to users or the vote creator.

**Comment by ghost** [comment]
> I guess that the process could be tuned to eliminate the requirement of a centralized server:
> 
> Optimal implementation:
> Step 1: The creator of voting is buying some CLO.
> 
> Step 2: The creator makes voting (sets how many users are going to participate, adds all the `ADDRESSES` for those users, and creates the question and options for the voting).
> 
> Step 3: The system automatically whitelists the `ADDRESSES` proposed at Step 2. ( only whitelisted address can vote).
> 
> Step 4: The system automatically sends CLO enough for the user to participate in the voting to every whitelisted address.
> 
> This requires users to create addresses and ensure privacy (by setting a set of rules for the voting that will prevent them from knowing who owns which address) at their own. Unfortunately, the smart-contract can not handle this in a proper way so the only "secure" approach is to delegate privacy/address generation to users or the vote creator.

I realize thought that it would be possible to create a smart contract that could create new addresses for each email added in the vote. 

As long as the creator can not see the vote that a certain address have made i suppose this would be good from a privacy perspective. 

Not as user friendly for a non crypto person at all though, 

I imagined that a voter would not have to know anything about how crypto works to take part in this. 

He would just receive an email with his new one time CLO address,, (used just for this vote(not for storing funds))
Instructions saying that he would need to vote within a certain time period and how to do it. 

Somehow confirm his vote and acknowledge that it is correct and he has not been compromised in any way.

Regarding centralization:
What about using IPFS servers for address creation and delivery of those to each email?

**Comment by KryptoSoulKing** [comment]
If the problem of centrality can be solved, i think that a voting system would be used even if there are not many possibilities at the beginning or more likely the possibilities will slowly show up, for example it could be used for my idea at the proposal from spigfish #87(lottery)

imo it could also be adopted by other projects

**Comment by shivneja** [comment]
> I guess that the process could be tuned to eliminate the requirement of a centralized server:
> 
> Optimal implementation:
> Step 1: The creator of voting is buying some CLO.
> 
> Step 2: The creator makes voting (sets how many users are going to participate, adds all the `ADDRESSES` for those users, and creates the question and options for the voting).
> 
> Step 3: The system automatically whitelists the `ADDRESSES` proposed at Step 2. ( only whitelisted address can vote).
> 
> Step 4: The system automatically sends CLO enough for the user to participate in the voting to every whitelisted address.
> 
> This requires users to create addresses and ensure privacy (by setting a set of rules for the voting that will prevent them from knowing who owns which address) at their own. Unfortunately, the smart-contract can not handle this in a proper way so the only "secure" approach is to delegate privacy/address generation to users or the vote creator.

@Dexaran
Can this also go cross-chain? Do you think it’s feasible? So the whitelisted addresses are maintained and approved by (say) the MYID project but the smart contract for voting is on Callisto?

**Comment by shivneja** [comment]
https://cryptocoingrowth.com/2020/02/16/indias-election-commission-developing-blockchain-system-for-voting/

Game on!!

Edit:
https://www.cnet.com/features/this-could-be-microsofts-most-important-product-in-2020-if-it-works/

@spigfish your competitors are making their moves.

Edit:
https://twitter.com/mincircle6/status/1230460502947454977?s=21
